customerio_test.go

```diff
@@ -1,155 +1,298 @@
-package customerio
+package customerio_test
 
 import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
 	"os"
+	"strconv"
+	"strings"
 	"testing"
-	"time"
+
+	"github.com/customerio/go-customerio"
 )
 
-var cio *CustomerIO
+var cio *customerio.CustomerIO
 
 func TestMain(m *testing.M) {
-
-	siteID := os.Getenv("CUSTOMERIO_SITE_ID")
-	apiKey := os.Getenv("CUSTOMERIO_API_KEY")
-
-	var exitCode int
-
-	if siteID != "" && apiKey != "" {
-		cio = NewCustomerIO(siteID, apiKey)
-		exitCode = m.Run()
-	} else {
-		exitCode = 1
-		fmt.Println("Must set CUSTOMERIO_SITE_ID and CUSTOMERIO_API_KEY environment variables to test this library")
+	srv := httptest.NewServer(http.HandlerFunc(handler))
+	defer srv.Close()
+	u, err := url.Parse(srv.URL)
+	if err != nil {
+		panic(err)
 	}
 
-	os.Exit(exitCode)
-}
-
-func TestIdentify(t *testing.T) {
+	cio = customerio.NewCustomerIO("siteid", "apikey")
+	cio.Host = u.Host
+	cio.SSL = false
```

Comment:
might be a good idea to comment this to make it clear that this is not desirable for production usage.